links-tests/Test/Xrefcheck/FtpLinks.hs

```diff
@@ -0,0 +1,84 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Test.Xrefcheck.FtpLinks
+  ( FtpHostOpt(..)
+  , ftpOptions
+  , test_FtpLinks
+  ) where
+
+import Data.Tagged (Tagged, untag)
+import Options.Applicative (help, long, strOption)
+import Test.Tasty (TestTree, askOption, testGroup)
+import Test.Tasty.HUnit (assertBool, assertFailure, testCase, (@?=))
+import Test.Tasty.Options as Tasty (IsOption (..), OptionDescription (Option), safeRead)
+
+import Xrefcheck.Config (Config (cVerification), VerifyConfig, defConfig)
+import Xrefcheck.Core (Flavor (GitHub))
+import Xrefcheck.Verify (VerifyError (..), checkExternalResource, verifyErrors, verifyOk)
+
+import Test.Orphans ()
+
+-- | A list with all the options needed to configure FTP links tests.
+ftpOptions :: [OptionDescription]
+ftpOptions =
+  [ Tasty.Option (Proxy @FtpHostOpt)
+  ]
+
+-- | Option specifying FTP host.
+newtype FtpHostOpt = FtpHostOpt Text
+  deriving (Show, Eq)
+
+instance IsOption FtpHostOpt where
+  defaultValue = FtpHostOpt "ftp://localhost"
+  optionName = "ftp-host"
+  optionHelp = "[Test.Xrefcheck.FtpLinks] FTP host without trailing slash"
+  parseValue v = FtpHostOpt <$> safeRead v
+  optionCLParser = FtpHostOpt <$> strOption
+    (  long (untag (optionName :: Tagged FtpHostOpt String))
+    <> help (untag (optionHelp :: Tagged FtpHostOpt String))
+    )
+
+
+config :: VerifyConfig
+config = cVerification $ defConfig GitHub
+
+test_FtpLinks :: TestTree
+test_FtpLinks = askOption $ \(FtpHostOpt host) -> do
+  let notRaisedMsg = "expected exceptions was not raised"
+
+  testGroup "Ftp links handler"
+    [ testCase "handles correct link to file" $ do
+        let link = host <> "/pub/file_exists.txt"
+        result <- checkExternalResource config link
+        verifyOk result @?= True
```

Comment:
If `checkExternalResource` returns some errors, then comparing two booleans using `verifyOk result @?= True` will display a very generic, and not really helpful message in the test logs, something like "Expected True, got False"

It would be better to compare the full data type, e.g. `result @?= VerifyResult []`.
That way, if the response contains some errors, those errors will be displayed in the test logs as well, giving the reader some idea of _why_ the test failed.

Reply:
Good idea, you're right

now instead of "Expected True, got False" I can see 
```
expected: VerifyResult []
         but got: VerifyResult [FileDoesNotExist "pub/file_exists.txt"]
```
for missing file